picard/formats/__init__.py

```diff
@@ -39,17 +40,47 @@ def supported_formats():
     return formats
 
 
+def guess_format(filename, options=_formats):
+    """Select the best matching file type amongst supported formats."""
+    results = []
+    with file(filename, "rb") as fileobj:
```

Comment:
We already have an open() method defined in the file. Which is why I opted to use file()